src/featurestorebundle/databricks/feature/writer/DatabricksFeatureStoreExistenceChecker.py

```diff
@@ -0,0 +1,38 @@
+from featurestorebundle.delta.TableExistenceChecker import TableExistenceChecker
+from featurestorebundle.databricks.DatabricksFeatureStoreClientFactory import DatabricksFeatureStoreClientFactory
+
+
+class DatabricksFeatureStoreExistenceChecker:
+    def __init__(
+        self,
+        table_existence_checker: TableExistenceChecker,
+        feature_store_client_factory: DatabricksFeatureStoreClientFactory,
+    ):
+        self.__table_existence_checker = table_existence_checker
+        self.__feature_store_client = feature_store_client_factory.create()
+
+    def exists(self, full_table_name: str) -> bool:
+        if self.__table_exists_in_feature_store(full_table_name) and not self.__table_exists_in_hive(full_table_name):
+            raise Exception(f"Table {full_table_name} exists in Databricks Feature Store but not in hive")
+
+        if not self.__table_exists_in_feature_store(full_table_name) and self.__table_exists_in_hive(full_table_name):
+            raise Exception(f"Table {full_table_name} exists in hive but not in Databricks Feature Store")
+
+        if self.__table_exists_in_feature_store(full_table_name) and self.__table_exists_in_hive(full_table_name):
+            return True
+
+        if not self.__table_exists_in_feature_store(full_table_name) and not self.__table_exists_in_hive(full_table_name):
+            return False
+
+        raise Exception(f"Databricks Feature store table {full_table_name} is in inconsistent state")
+
+    def __table_exists_in_feature_store(self, full_table_name: str) -> bool:
+        try:
+            self.__feature_store_client.get_feature_table(full_table_name)
```

Comment:
Just to point out: The new release 0.5.0 of DBX FS Python API has this method renamed to `get_table` only.

Reply:
Just check this please, according to [this link](https://docs.databricks.com/applications/machine-learning/feature-store/python-api.html) in ML DBR 10.4 the 0.5.0 version is used.